pkg/controller/blockdevice/scanner.go

```diff
@@ -216,8 +233,21 @@ func (s *Scanner) SaveBlockDevice(bd *diskv1.BlockDevice, autoProvisioned bool)
 		bd.Spec.FileSystem.ForceFormatted = true
 		bd.Spec.FileSystem.Provisioned = true
 	}
-	logrus.Infof("Add new block device %s with device: %s", bd.Name, bd.Spec.DevPath)
-	return s.Blockdevices.Create(bd)
+	curBd, err := s.Blockdevices.Get(bd.Namespace, bd.Name, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			if autoProvisioned {
+				bd.Spec.FileSystem.ForceFormatted = true
+				bd.Spec.FileSystem.Provisioned = true
+			}
+			logrus.Infof("Add new block device %s with device: %s", bd.Name, bd.Spec.DevPath)
+			return s.Blockdevices.Create(bd)
+		}
+		return nil, err
+	}
+	logrus.Infof("The inactive block device %s is coming back", bd.Name)
+	curBd.Status.State = diskv1.BlockDeviceActive
```

Comment:
this may be an edge case.. but may happen in real life. remove a disk and add a new disk. the disk slot identifier is the same... say /dev/sdc, however this is a new disk, and we are just marking the disk as active.

in my tests on libvirt, i added a disk /dev/sdc, with a wwn, which was duely processed and added to a blockdevice.

I removed this disk, and deleted the disk, and added a new disk, which would be similar to replacing a disk in the same host slot.

the new disk gets the same device identifier /dev/sdc, but the underlying disk is different, and has a different wwn, but is processed as an updated and device status does not match the physical disk.

should be add more checks here like 
```suggestion
    if curBd.Status.State.Details.WWN != bd.Status.State.Details.WWN {
        // delete and create a new BD so it can fall through the new block device processing loop
    }
```

Reply:
closing this.. could not re-produce this issue.

Reply:
Thanks @ibrokethecloud for taking care of this!
I add some logs to record the `wwn` if we meet this again.